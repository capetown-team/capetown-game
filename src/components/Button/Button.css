.Button {
    width: 280px;
    height: 38px;
    background: #6995E0;
    vertical-align: middle;
    text-align: center;
    display: block;
    margin: auto;
    margin-top: 10px;
}