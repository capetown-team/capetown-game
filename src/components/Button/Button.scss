@import '../../styles/layout/colors';

.button {
    background-color: $light-blue;
    color: $white;
    text-align: center;
    border-radius: 3px;
    border: none;

    &_size {
        &_s {
            padding: 7px 15px;
        }

        &_m {
            display: block;
            padding: 10px 0;
            width: 100%;
        }

        &_l {
            display: block;
            width: 336px;
            height: 46px;
        }
    }

    &:hover {
        background-color: rgba($light-blue, .9);
    }

    &:active {
        box-shadow: inset 0 3px 5px rgba($black, .3);
    }

    &:disabled {
        background-color: $grey;
    }
}
