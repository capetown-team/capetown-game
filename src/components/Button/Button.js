import React from 'react';

import './Button.css';

export class Button extends React.Component {
    render() {
        const { onClick, children } = this.props;
        return ( < button className = "Button"
            onClick = { onClick } > { children } <
            /button>
        );
    }
}
export default Button;