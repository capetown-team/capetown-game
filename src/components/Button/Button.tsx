import React, { memo, FC, MouseEvent, ReactNode } from 'react';
import block from 'bem-cn-lite';

import './Button.scss';

const b = block('button');

type Props = {
  placeholder?: string;
  onClick?: (event: MouseEvent) => void;
  size: string;
  type: string;
  children: ReactNode | string;
  disabled: boolean;
};

const Button: FC<Props> = ({ size, children, onClick, disabled }) => {
  const className = b({ size });
  return (
    <button
      disabled={disabled}
      type="submit"
      className={className}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

const WrappedButton = memo(Button);

export { WrappedButton as Button };
