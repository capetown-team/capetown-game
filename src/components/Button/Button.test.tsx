import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { Button } from './Button';

describe('<Button />', () => {
  it('should renders correct <Button />', () => {
    const wrapper = shallow(<Button size="s">test</Button>);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
