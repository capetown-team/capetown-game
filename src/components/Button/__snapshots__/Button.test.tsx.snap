// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should renders correct <Button /> 1`] = `
<button
  className="button button_size_s"
  type="button"
>
  test
</button>
`;
